sds/src/secondary_validation/lithuanian_personal_identification_number_checksum.rs

```diff
@@ -0,0 +1,72 @@
+use crate::secondary_validation::Validator;
+
+const ROUND_1_WEIGHTS: &[u32] = &[1, 2, 3, 4, 5, 6, 7, 8, 9, 1];
+const ROUND_2_WEIGHTS: &[u32] = &[3, 4, 5, 6, 7, 8, 9, 1, 2, 3];
+
+pub struct LithuanianPersonalIdentificationNumberChecksum;
+
+impl Validator for LithuanianPersonalIdentificationNumberChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if regex_match.chars().any(|c| !c.is_ascii_digit()) {
+            return false;
+        }
+
+        // 10 chars + 1 checksum
+        if regex_match.len() != 11 {
+            return false;
+        }
+
+        let mut checksum = run_round(regex_match, ROUND_1_WEIGHTS) % 11;
+        if checksum == 10 {
+            checksum = run_round(regex_match, ROUND_2_WEIGHTS) % 11;
+            if checksum == 10 {
+                checksum = 0;
+            }
+        }
+
+        let actual_checksum = regex_match.chars().last().unwrap().to_digit(10).unwrap();
+
+        checksum == actual_checksum
+    }
+}
+
+fn run_round(value: &str, weights: &[u32]) -> u32 {
+    let mut checksum = 0;
+
+    for (i, c) in value.chars().take(10).enumerate() {
+        checksum += weights[i] * c.to_digit(10).unwrap();
+    }
+
+    checksum
```

Comment:
If the input is converted to digits right away, then a slice to be used as the value as it should simplify this code:

```suggestion
fn run_round(value: &[u32], weights: &[u32; 10]) -> u32 {
    let mut checksum = 0;
    for (digit, weight) in value.iter().zip(weights) {
        checksum += digit * weight;
    }
    checksum
```